File: samples/Sample.lua
-- Islands Dark Theme - Lua Sample

local MAX_RETRIES = 3
local VERSION = "1.0.0"

local Color = {}
Color.__index = Color

function Color.new(r, g, b)
    return setmetatable({ r = r, g = g, b = b }, Color)
end

function Color:toHex()
    return string.format("#%02X%02X%02X", self.r, self.g, self.b)
end

function Color.fromHex(hex)
    hex = hex:gsub("^#", "")
    local r = tonumber(hex:sub(1, 2), 16)
    local g = tonumber(hex:sub(3, 4), 16)
    local b = tonumber(hex:sub(5, 6), 16)
    return Color.new(r, g, b)
end

local Theme = {}
Theme.__index = Theme

function Theme.new(name)
    local self = setmetatable({}, Theme)
    self.name = name
    self.colors = {}
    return self
end

function Theme:addColor(key, color)
    self.colors[key] = color
end

function Theme:getColor(key)
    return self.colors[key]
end

function Theme:isDark()
    return true
end

function Theme:printInfo()
    print("Theme: " .. self.name)
    for key, color in pairs(self.colors) do
        print(string.format("  %s: %s", key, color:toHex()))
    end
end

local function processItems(items, prefix)
    prefix = prefix or "item"
    local result = {}
    for i, item in ipairs(items) do
        table.insert(result, string.format("%s_%d: %s", prefix, i, item))
    end
    return result
end

local theme = Theme.new("Islands Dark")
theme:addColor("keyword", Color.fromHex("#CF8E6D"))
theme:addColor("background", Color.new(25, 26, 28))
theme:printInfo()

local items = processItems({ "alpha", "beta", "gamma" })
for _, item in ipairs(items) do
    print("  " .. item)
end

return Theme

File: samples/Sample.rs
use std::collections::HashMap;
use std::fmt;

const MAX_RETRIES: usize = 3;
const VERSION: &str = "1.0.0";

struct Color {
    r: u8,
    g: u8,
    b: u8,
}

impl Color {
    fn new(r: u8, g: u8, b: u8) -> Self {
        Self { r, g, b }
    }

    fn from_hex(hex: &str) -> Self {
        let hex = hex.trim_start_matches('#');
        let r = u8::from_str_radix(&hex[0..2], 16).unwrap_or(0);
        let g = u8::from_str_radix(&hex[2..4], 16).unwrap_or(0);
        let b = u8::from_str_radix(&hex[4..6], 16).unwrap_or(0);
        Self { r, g, b }
    }

    fn to_hex(&self) -> String {
        format!("#{:02X}{:02X}{:02X}", self.r, self.g, self.b)
    }
}

impl fmt::Display for Color {
    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
        write!(f, "rgb({}, {}, {})", self.r, self.g, self.b)
    }
}

#[derive(Debug)]
struct Theme {
    name: String,
    colors: HashMap<String, Color>,
}

impl Theme {
    fn new(name: &str) -> Self {
        Self {
            name: name.to_string(),
            colors: HashMap::new(),
        }
    }

    fn add_color(&mut self, key: &str, color: Color) {
        self.colors.insert(key.to_string(), color);
    }

    fn get_color(&self, key: &str) -> Option<&Color> {
        self.colors.get(key)
    }
}

enum ThemeError {
    NotFound(String),
    InvalidFormat(String),
}

fn process_items(items: &[&str], prefix: &str) -> Vec<String> {
    items
        .iter()
        .enumerate()
        .map(|(i, item)| format!("{}_{}: {}", prefix, i, item))
        .collect()
}

fn main() {
    let mut theme = Theme::new("Islands Dark");
    theme.add_color("background", Color::new(25, 26, 28));
    theme.add_color("keyword", Color::from_hex("#CF8E6D"));
    theme.add_color("string", Color::from_hex("#6AAB73"));

    println!("Theme: {}", theme.name);

    let items = ["alpha", "beta", "gamma"];
    for item in process_items(&items, "item") {
        println!("  {item}");
    }
}